openflights.js

```diff
@@ -106,6 +106,10 @@ function init(){
 
   map = new OpenLayers.Map('map', {
     center: new OpenLayers.LonLat(0, 1682837.6144925),
+	attribution: {
+		title: "&copy; OpenStreetMap contributors",
+		href: "https://www.openstreetmap.org/"
+	},
```

Comment:
I think, looking at line 134 below, this needs to be under a layer...

See also like https://github.com/openlayers/openlayers/compare/release-2.11...v2.13.1#diff-99e6b60d1f8e64080689545ad1091989875914a0d2b7bde885525413682473beR13-R17

And maybe an attribution control adding?

Reply:
I'm also curious which layer(s) the OSM attribution needs to be against too. Especially as below the "Artistic", "Political" and "Satellite" have attribution...

But I also cannot see where those attributions appear on the maps either.

Reply:
Ok, and the key is for those attributions not appearing...

https://github.com/openlayers/openlayers/blob/release-2.11/examples/attribution.html#L31-L34

![Screenshot 2023-04-23 at 16 40 00](https://user-images.githubusercontent.com/67615/233849580-ab7e9ea6-9b3f-4b9f-aa38-b51634abdc25.png)

It's not pretty... But it's a start...

```
diff --git a/openflights.js b/openflights.js
index 8c2db2d..9e43d17 100644
--- a/openflights.js
+++ b/openflights.js
@@ -115,7 +115,8 @@ function init(){
       new OpenLayers.Control.Navigation({'title': gt.gettext("Toggle pan and region select mode")}),
       new OpenLayers.Control.LayerSwitcher({'ascending':false, 'title': gt.gettext('Switch map layers')}),
       new OpenLayers.Control.ScaleLine(),
-      new OpenLayers.Control.OverviewMap({'title': gt.gettext("Toggle overview map")})
+      new OpenLayers.Control.OverviewMap({'title': gt.gettext("Toggle overview map")}),
+      new OpenLayers.Control.Attribution()
     ] });
 
   // Horrible hack to stop OpenLayers 2 from showing ZL < 2
```

Need to play around with the location of things...

Reply:
Which looks to be done in CSS - https://stackoverflow.com/questions/20981679/openlayers-map-change-position-of-attribution-in-javascript

````
.olControlAttribution {
    font-size: smaller; 
    right: 3px; 
    bottom: 4.5em; 
    position: absolute; 
    display: block;
}
```

Reply:
Hmm. And after https://github.com/jpatokal/openflights/commit/a5c352206c7d31fe59ad2349a34c3b4597dd5a15 it seems to be working as expected...

Yeah, the colours suck on some of the maps...

![Screenshot 2023-04-23 at 17 03 33](https://user-images.githubusercontent.com/67615/233850856-f7e54acb-f873-4c63-9764-05ec39cc4502.png)
![Screenshot 2023-04-23 at 17 03 36](https://user-images.githubusercontent.com/67615/233850858-e404f881-af0f-4c96-8905-0ea2439a5530.png)
![Screenshot 2023-04-23 at 17 03 39](https://user-images.githubusercontent.com/67615/233850860-82bcd93d-5105-4d5a-9f2e-9e79a4d9185f.png)